cosmic-app-list/src/app.rs

```diff
@@ -593,6 +993,40 @@ impl Application for CosmicAppList {
                         Some(Message::RemovedSeat(seat))
                     }
                 },
+                // XXX Must be done to catch a finished drag after the source is removed
+                // (for now, the source is removed when the drag starts)
+                cosmic::iced_native::Event::PlatformSpecific(
+                    cosmic::iced_native::event::PlatformSpecific::Wayland(
+                        cosmic::iced_sctk::event::wayland::Event::DataSource(
+                            cosmic::iced_sctk::event::wayland::DataSourceEvent::DndFinished
+                            | cosmic::iced_sctk::event::wayland::DataSourceEvent::Cancelled,
+                        ),
+                    ),
+                ) => Some(Message::DragFinished),
+                cosmic::iced_native::Event::PlatformSpecific(
+                    cosmic::iced_native::event::PlatformSpecific::Wayland(
+                        cosmic::iced_native::event::wayland::Event::DndOffer(
+                            cosmic::iced_native::event::wayland::DndOfferEvent::Enter {
+                                mime_types,
+                                ..
+                            },
+                        ),
+                    ),
+                ) => {
+                    if mime_types.iter().any(|m| m == MIME_TYPE) {
+                        Some(Message::StartListeningForDnd)
+                    } else {
+                        None
+                    }
+                }
+                cosmic::iced_native::Event::PlatformSpecific(
+                    cosmic::iced_native::event::PlatformSpecific::Wayland(
+                        cosmic::iced_native::event::wayland::Event::DndOffer(
+                            cosmic::iced_native::event::wayland::DndOfferEvent::Leave
+                            | cosmic::iced_native::event::wayland::DndOfferEvent::DropPerformed,
+                        ),
+                    ),
```

Comment:
Match cases taking this many lines are... quite something.

Maybe add an import for `cosmic::iced_native::event` to the file so these could be a bit shorter?

Reply:
Ok ya lol :sweat_smile: